R/download_dataset.R

```diff
@@ -0,0 +1,170 @@
+#' Download data for publication
+#'
+#' ...
+#'
+#' @param connection A valid connection to the ETN database.
+#' @param animal_project_code (string) Animal project to download data from.
+#' @param directory (string) Path to local download directory.
+#' @param scientific_name (string) One or more scientific names to filter on.
+#'
+#' @return ...
+#'
+#' @export
+#'
+#' @importFrom glue glue
+#' @importFrom dplyr %>% arrange distinct filter group_by mutate pull select tibble
+#'
+#' @examples
+#' \dontrun{
+#' ...
+#' }
+download_dataset <- function(connection = con,
+                        animal_project_code = NULL,
+                        directory = animal_project_code,
+                        scientific_name = NULL) {
+  # Check connection
+  check_connection(connection)
+
+  # Check animal_project_code
+  if (is.null(animal_project_code)) {
+    stop("Please provide an animal_project_code")
+  } else {
+    valid_animal_project_codes <- list_animal_project_codes(connection)
+    check_value(animal_project_code, valid_animal_project_codes, "animal_project_code")
+  }
+
+  # Check directory
+  if (!dir.exists(directory)) {
+    dir.create(directory)
+  }
```

Comment:
Based on a comment of Thierry while developing `setup_codingclub_session()` I learned that we can avoid the if. See https://github.com/inbo/inborutils/blob/master/R/setup_codingclub_session.R#L129

Reply:
Resolved